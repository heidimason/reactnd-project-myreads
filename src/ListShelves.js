import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import ListBooks from './ListBooks'
import * as strMethods from './utils/strMethods'

class ListShelves extends Component {
	render() {
        const { bookshelves, books } = this.props

		return (
			<div>
            	<ul className="list-books-content">
            		{bookshelves.map( (bookshelf, index) => (
	            		<li className="bookshelf" key={index}>
	              			<h2 className="bookshelf-title">{bookshelf.name}</h2>

	                        <ListBooks
                                showingBooks={books.filter( book => strMethods.camelize( bookshelf.name.toLowerCase() ) === book.shelf )}
                            />
	                    </li>
                    ))}
                </ul>

                <div className="open-search">
                    <Link to="/search">Add a book</Link>
                </div>
          	</div>
		)
	}
}

export default ListShelves