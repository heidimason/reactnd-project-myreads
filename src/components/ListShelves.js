import React from 'react'
import { Link } from 'react-router-dom'
import ListBooks from './ListBooks'
import * as helpers from '../utils/helpers'

function ListShelves (props) {
    const { bookshelves, shelvedBooks, onMoveBook } = props

	return (
		<div>
        	<ul className="list-books-content">
        		{bookshelves.map( (bookshelf, index) => (
            		<li className="bookshelf" key={index}>
              			<h2 className="bookshelf-title">{bookshelf.name}</h2>

                        <ListBooks
                            showingBooks={shelvedBooks.filter( book => helpers.camelize( bookshelf.name.toLowerCase() ) === book.shelf )}
                            onMoveBook={onMoveBook}
                        />
                    </li>
                ))}
            </ul>

            <div className="open-search">
                <Link to="/books/search">Add a book</Link>
            </div>
      	</div>
	)
}

export default ListShelves