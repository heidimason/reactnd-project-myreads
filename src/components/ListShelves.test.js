import React from 'react'
import renderer from 'react-test-renderer'
import ListShelves from './ListShelves'
import { BrowserRouter, Link } from 'react-router-dom'

describe('<ListShelves />', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(
				<BrowserRouter>
					<Link to="/myreads/search">Add a book</Link>
				</BrowserRouter>
			)
			.toJSON()

		expect(tree).toMatchSnapshot()
	})
})