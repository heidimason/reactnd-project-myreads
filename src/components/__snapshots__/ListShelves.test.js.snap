exports[`<ListShelves /> renders correctly 1`] = `
<a
  href="/myreads/search"
  onClick={[Function]}>
  Add a book
</a>
`;
